### ISSUE TITLE
Error in Step 2 Fine tuning.

### ISSUE THREAD

**Issue by mercuryyy** [erew123/alltalk_tts#239]
Getting an error in step 2 of fine tuning.

I am also going to post Step 1 info for debugging. 

my training wav file is 2 min and 22 seconds long.

After step 1 finishes in the tmp-trn/wavs i only see 1 file "anelvoice_00000000.wav"
I thought it would split the file into many parts" 

Anyway step 1 completes and then i get an error in step 2. 

Step 1:

```
[FINETUNE] [94mPart of AllTalk[0m https://github.com/erew123/alltalk_tts/
[FINETUNE] [94mCoqui Public Model License[0m
[FINETUNE] [94mhttps://coqui.ai/cpml.txt[0m
[FINETUNE] [94mWhisper model: [92mlarge-v2 [94mLanguage: [92men [94mEvaluation data percentage: [92m15.0%[0m
[FINETUNE] [94mStarting Step 1[0m - Preparing Audio/Generating the dataset
[FINETUNE] Updated lang.txt with the target language.
[FINETUNE] Loading Whisper Model: large-v2
[FINETUNE] Model will be downloaded if its not available, which will take a few minutes.
[FINETUNE] Current working file: /home/izzy/alltalk/alltalk2/alltalk_tts/finetune/put-voice-samples-in-here/anelvoice.wav
[FINETUNE] Processing audio with duration 02:22.571
[FINETUNE] VAD filter removed 00:05.072 of audio
[FINETUNE] Train CSV: /home/izzy/alltalk/alltalk2/alltalk_tts/finetune/tmp-trn/metadata_train.csv
[FINETUNE] Eval CSV: /home/izzy/alltalk/alltalk2/alltalk_tts/finetune/tmp-trn/metadata_eval.csv
[FINETUNE] Audio Total: 142.5705442176871
[FINETUNE] Dataset Generated. Move to Step 2

```

```
Running on local URL:  http://127.0.0.1:7052

To create a public link, set `share=True` in `launch()`.
IMPORTANT: You are using gradio version 3.50.2, however version 4.29.0 is available, please upgrade.
--------
[FINETUNE] [94mPart of AllTalk[0m https://github.com/erew123/alltalk_tts/
[FINETUNE] [94mCoqui Public Model License[0m
[FINETUNE] [94mhttps://coqui.ai/cpml.txt[0m
[FINETUNE] [94mWhisper model: [92mlarge-v2 [94mLanguage: [92men [94mEvaluation data percentage: [92m15.0%[0m
[FINETUNE] [94mStarting Step 1[0m - Preparing Audio/Generating the dataset
[FINETUNE] Updated lang.txt with the target language.
[FINETUNE] Loading Whisper Model: large-v2
[FINETUNE] Model will be downloaded if its not available, which will take a few minutes.
[FINETUNE] Current working file: /home/izzy/alltalk/alltalk2/alltalk_tts/finetune/put-voice-samples-in-here/anelvoice.wav
[FINETUNE] Processing audio with duration 02:22.571
[FINETUNE] VAD filter removed 00:05.072 of audio
[FINETUNE] Train CSV: /home/izzy/alltalk/alltalk2/alltalk_tts/finetune/tmp-trn/metadata_train.csv
[FINETUNE] Eval CSV: /home/izzy/alltalk/alltalk2/alltalk_tts/finetune/tmp-trn/metadata_eval.csv
[FINETUNE] Audio Total: 142.5705442176871
[FINETUNE] Dataset Generated. Move to Step 2
[FINETUNE] [94mStarting Step 2[0m - Fine-tuning the XTTS Encoder
[FINETUNE] [94mLanguage: [92men [94mEpochs: [92m10 [94mBatch size: [92m4[0m [94mGrad accumulation steps: [92m1[0m
[FINETUNE] [94mTraining   : [92m/home/izzy/alltalk/alltalk2/alltalk_tts/finetune/tmp-trn/metadata_train.csv[0m
[FINETUNE] [94mEvaluation : [92m/home/izzy/alltalk/alltalk2/alltalk_tts/finetune/tmp-trn/metadata_eval.csv[0m
[FINETUNE] [94mAvailable VRAM: [92m23.53 GB[0m
[FINETUNE] Starting finetuning on [92mBase Model[0m
>> DVAE weights restored from: /home/izzy/alltalk/alltalk2/alltalk_tts/models/xttsv2_2.0.2/dvae.pth
 | > Found 1 files in /home/izzy/alltalk/alltalk2/alltalk_tts/finetune/tmp-trn
 > Sampling by language: dict_keys(['en'])
Traceback (most recent call last):
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/trainer/trainer.py", line 1833, in fit
    self._fit()
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/trainer/trainer.py", line 1785, in _fit
    self.train_epoch()
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/trainer/trainer.py", line 1503, in train_epoch
    for cur_step, batch in enumerate(self.train_loader):
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 631, in __next__
    data = self._next_data()
           ^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 1346, in _next_data
    return self._process_data(data)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 1372, in _process_data
    data.reraise()
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/_utils.py", line 705, in reraise
    raise exception
RecursionError: Caught RecursionError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/_utils/worker.py", line 308, in _worker_loop
    data = fetcher.fetch(index)  # type: ignore[possibly-undefined]
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/_utils/fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/_utils/fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
            ~~~~~~~~~~~~^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/TTS/tts/layers/xtts/trainer/dataset.py", line 180, in __getitem__
    return self[1]
           ~~~~^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/TTS/tts/layers/xtts/trainer/dataset.py", line 156, in __getitem__
    return self[1]
           ~~~~^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/TTS/tts/layers/xtts/trainer/dataset.py", line 156, in __getitem__
    return self[1]
           ~~~~^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/TTS/tts/layers/xtts/trainer/dataset.py", line 156, in __getitem__
    return self[1]
           ~~~~^^^
  [Previous line repeated 967 more times]
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/TTS/tts/layers/xtts/trainer/dataset.py", line 146, in __getitem__
    index = random.randint(0, len(self.samples[lang]) - 1)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/random.py", line 362, in randint
    return self.randrange(a, b+1)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/random.py", line 344, in randrange
    return istart + self._randbelow(width)
                    ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/random.py", line 239, in _randbelow_with_getrandbits
    k = n.bit_length()  # don't use (n-1) here because n can be 1
        ^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded while calling a Python object

Traceback (most recent call last):
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/trainer/trainer.py", line 1833, in fit
    self._fit()
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/trainer/trainer.py", line 1785, in _fit
    self.train_epoch()
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/trainer/trainer.py", line 1503, in train_epoch
    for cur_step, batch in enumerate(self.train_loader):
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 631, in __next__
    data = self._next_data()
           ^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 1346, in _next_data
    return self._process_data(data)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 1372, in _process_data
    data.reraise()
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/_utils.py", line 705, in reraise
    raise exception
RecursionError: Caught RecursionError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/_utils/worker.py", line 308, in _worker_loop
    data = fetcher.fetch(index)  # type: ignore[possibly-undefined]
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/_utils/fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/torch/utils/data/_utils/fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
            ~~~~~~~~~~~~^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/TTS/tts/layers/xtts/trainer/dataset.py", line 180, in __getitem__
    return self[1]
           ~~~~^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/TTS/tts/layers/xtts/trainer/dataset.py", line 156, in __getitem__
    return self[1]
           ~~~~^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/TTS/tts/layers/xtts/trainer/dataset.py", line 156, in __getitem__
    return self[1]
           ~~~~^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/TTS/tts/layers/xtts/trainer/dataset.py", line 156, in __getitem__
    return self[1]
           ~~~~^^^
  [Previous line repeated 967 more times]
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/TTS/tts/layers/xtts/trainer/dataset.py", line 146, in __getitem__
    index = random.randint(0, len(self.samples[lang]) - 1)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/random.py", line 362, in randint
    return self.randrange(a, b+1)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/random.py", line 344, in randrange
    return istart + self._randbelow(width)
                    ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/random.py", line 239, in _randbelow_with_getrandbits
    k = n.bit_length()  # don't use (n-1) here because n can be 1
        ^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded while calling a Python object


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/finetune.py", line 1523, in train_model
    config_path, original_xtts_checkpoint, vocab_file, exp_path, speaker_wav = train_gpt(language, num_epochs, batch_size, grad_acumm, train_csv, eval_csv, learning_rate, output_path=str(output_path), max_audio_length=max_audio_length)
                                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/finetune.py", line 691, in train_gpt
    trainer.fit()
  File "/home/izzy/alltalk/alltalk2/alltalk_tts/alltalk_environment/env/lib/python3.11/site-packages/trainer/trainer.py", line 1862, in fit
    sys.exit(1)
SystemExit: 1
```

**Comment by erew123** [comment]
Hi @mercuryyy  

Id be pretty confident it couldn't split your wav file down properly. With the file being so small, I cant recall if you split it into 2x files before running Step 1, if it will accept that or not (too many revisions of code to think through to recall exactly).

However in the next release (well the BETA) Ive added extra code to force files to be split in the situation where it struggles with doing this. Hopefully I will have that out within the next 24-48 hours.

For now, you could try splitting the audio in to two parts with Audacity and see if that clears it.

![image](https://github.com/erew123/alltalk_tts/assets/35898566/a39950c2-c28d-46be-8342-8aff2f11e932)

Thanks

**Comment by mercuryyy** [comment]
thank you got it working by manually splitting the files.

Cant wait for the v2 Beta hope you release soon :)
